<a href="https://colab.research.google.com/github/gabrielcn/Python_Data_Science_Numpy/blob/main/python_data_science_e_numpy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Verificando a versão do python**

In [711]:
!python -V

Python 3.7.13


**1.3 Trabalhando com arrays numpy**

In [712]:
import numpy as np #Importar a biblioteca numpy

In [713]:
km = np.loadtxt('carros-km.txt') #carrega o arquivo txt dentro de um array numpy

In [714]:
km

array([4.44100e+04, 5.71200e+03, 3.71230e+04, 0.00000e+00, 2.57570e+04,
       1.07280e+04, 0.00000e+00, 7.75990e+04, 9.91970e+04, 3.79780e+04,
       1.28590e+04, 8.05200e+03, 8.97730e+04, 0.00000e+00, 4.14570e+04,
       1.15607e+05, 4.64490e+04, 0.00000e+00, 3.70860e+04, 1.51730e+04,
       1.01193e+05, 0.00000e+00, 9.80790e+04, 1.02959e+05, 0.00000e+00,
       0.00000e+00, 5.79500e+03, 0.00000e+00, 5.88480e+04, 9.43810e+04,
       3.01630e+04, 5.33320e+04, 1.77200e+04, 3.38080e+04, 9.06840e+04,
       4.39750e+04, 0.00000e+00, 0.00000e+00, 5.52600e+03, 0.00000e+00,
       9.34150e+04, 4.07620e+04, 0.00000e+00, 8.63020e+04, 0.00000e+00,
       9.75500e+03, 6.99450e+04, 2.39500e+03, 0.00000e+00, 8.03490e+04,
       8.55540e+04, 5.04960e+04, 6.77160e+04, 9.39470e+04, 3.53450e+04,
       8.10070e+04, 1.19513e+05, 0.00000e+00, 0.00000e+00, 0.00000e+00,
       0.00000e+00, 0.00000e+00, 1.18895e+05, 4.85090e+04, 1.00912e+05,
       9.56490e+04, 0.00000e+00, 9.04950e+04, 0.00000e+00, 2.913

In [715]:
anos = np.loadtxt('carros-anos.txt', dtype = int) #carrega o arquivo txt dentro de um array numpy e o dtype = int define que a variável "anos" é do tipo inteiro

In [716]:
anos

array([2003, 1991, 1990, 2019, 2006, 2012, 2019, 2009, 2010, 2011, 2002,
       2007, 2001, 2019, 2009, 2016, 2012, 2019, 2016, 2017, 2002, 2019,
       1994, 1991, 2019, 2019, 2016, 2019, 2002, 1991, 1997, 1995, 2018,
       1991, 2014, 1995, 2019, 2019, 2009, 2019, 2013, 2003, 2019, 2002,
       2019, 2002, 2004, 2001, 2019, 2003, 2009, 2006, 2000, 2006, 2010,
       1994, 2007, 2019, 2019, 2019, 2019, 2019, 2017, 2011, 2001, 1996,
       2019, 1994, 2019, 2007, 1994, 2007, 1997, 2006, 2013, 1994, 2008,
       2008, 2019, 2019, 1999, 2007, 1993, 1995, 2019, 2019, 2016, 2011,
       2019, 2002, 2013, 2017, 2016, 2008, 2018, 2003, 2003, 2004, 2010,
       2019, 2006, 2001, 2019, 2015, 2018, 2017, 1992, 2001, 1993, 1995,
       2003, 2014, 2008, 2013, 2019, 2017, 2019, 2019, 2006, 2015, 2007,
       2005, 2017, 1998, 2019, 1999, 2019, 2009, 2003, 1994, 2019, 1994,
       2019, 2015, 1996, 1997, 1993, 2019, 1992, 1995, 2019, 1990, 2019,
       2019, 1991, 1990, 2012, 2007, 2013, 2000, 20

**Obtendo a quilometragem média por ano**

In [717]:
km_media = km / (2022 - anos)

In [718]:
km_media

array([2.33736842e+03, 1.84258065e+02, 1.16009375e+03, 0.00000000e+00,
       1.60981250e+03, 1.07280000e+03, 0.00000000e+00, 5.96915385e+03,
       8.26641667e+03, 3.45254545e+03, 6.42950000e+02, 5.36800000e+02,
       4.27490476e+03, 0.00000000e+00, 3.18900000e+03, 1.92678333e+04,
       4.64490000e+03, 0.00000000e+00, 6.18100000e+03, 3.03460000e+03,
       5.05965000e+03, 0.00000000e+00, 3.50282143e+03, 3.32125806e+03,
       0.00000000e+00, 0.00000000e+00, 9.65833333e+02, 0.00000000e+00,
       2.94240000e+03, 3.04454839e+03, 1.20652000e+03, 1.97525926e+03,
       4.43000000e+03, 1.09058065e+03, 1.13355000e+04, 1.62870370e+03,
       0.00000000e+00, 0.00000000e+00, 4.25076923e+02, 0.00000000e+00,
       1.03794444e+04, 2.14536842e+03, 0.00000000e+00, 4.31510000e+03,
       0.00000000e+00, 4.87750000e+02, 3.88583333e+03, 1.14047619e+02,
       0.00000000e+00, 4.22889474e+03, 6.58107692e+03, 3.15600000e+03,
       3.07800000e+03, 5.87168750e+03, 2.94541667e+03, 2.89310714e+03,
      

In [719]:
type(km_media) #Verifica o tipo da variável, no caso a variável é do tipo numpy.ndarray

numpy.ndarray

**2. CARACTERÍSICAS BÁSICAS DA LINGUAGEM**

**2.1 Operações matemáticas**
    
    Operadores aritméticos: +, -, *, /, **, %, //

**Adição (+)**

In [720]:
2 + 2

4

**Subtração (-)**

In [721]:
2 - 2

0

**Multiplicação (*)**

In [722]:
2 * 2

4

**Divisão (/) e (//)**

A operação divisão sempre retorna um número de ponto flutuante

In [723]:
10 / 3

3.3333333333333335

In [724]:
10 // 3 #Retorna somente a parte inteira da divisão

3

**Exponenciação (****)

In [725]:
2 ** 3

8

**Resto da divisão (%)**

In [726]:
10 % 3

1

**Expressões matemáticas**

In [727]:
5 * 2 + 3 * 2

16

In [728]:
(5 * 2) + (3 * 2)

16

In [729]:
5 * (2 + 3) * 2

50

**A variável**

No modo interativo, o último resultado impresso é atribuído a variável _

In [730]:
5 * 2

10

In [731]:
_ + 3 * 2

16

In [732]:
_ / 2

8.0

**2.2 Variáveis**

**Nomes de variáveis**

° Nomes de varáveis podem começar com letras (a - z, A - Z) ou o caractere underscore (_):

Altura

_peso

° O restante de nome pode conter letras, números e o caractere "_":

nome_da_variavel

_valor

dia_28_11_

° Os nomes são case sensitive:

Nome_Da_Variavel ≠ nome_da_variavel ≠ NOME_DA_VARIAVEL

### <font color=red>Observações:
- Existem algumas palavras reservadas da linguagem que não podem ser utilizadas como nomes de variável:

| |Lista de palavras <br>reservadas em Python| |
|:-------------:|:------------:|:-------------:|
| and           | as           | not           | 
| assert        | finally      | or            | 
| break         | for          | pass          | 
| class         | from         | nonlocal      | 
| continue      | global       | raise         | 
| def           | if           | return        | 
| del           | import       | try           | 
| elif          | in           | while         | 
| else          | is           | with          | 
| except        | lambda       | yield         | 
| False         | True         | None          | 



### Declaração de variáveis

### Operadores de atribuição: $=$, $+=$, $-=$, $*=$, $/=$, $**=$, $\%=$, $//=$

In [733]:
ano_atual = 2022
ano_fabricacao = 2003
km_total = 44410.0

In [734]:
ano_atual

2022

In [735]:
ano_fabricacao

2003

In [736]:
km_total

44410.0

**Operações com variáveis**

```
# Isto está formatado como código
```




# $$km_{média} = \frac {km_{total}}{(Ano_{atual} - Ano_{fabricação})}$$


In [737]:
ano_atual = 2022
ano_fabricacao = 2003
km_total = 44410.0
km_media = km_total / (ano_atual - ano_fabricacao)
km_media

2337.3684210526317

In [738]:
ano_atual = 2022
ano_fabricacao = 2003
km_total = 44410.0
km_media = km_total / (ano_atual - ano_fabricacao)

km_total = km_total + km_media
km_total

46747.36842105263

### Conclusão:
```
"valor = valor + 1" é equivalente a "valor += 1"
```

### Declaração múltipla

In [739]:
ano_atual, ano_fabricacao, km_total = 2022, 2003, 44410.0

In [740]:
ano_atual

2022

In [741]:
ano_fabricacao

2003

In [742]:
km_total

44410.0

In [743]:
ano_atual, ano_fabricacao, km_total = 2022, 2003, 44410.0
km_media = km_total / (ano_atual - ano_fabricacao)
km_media

2337.3684210526317

**2.3 Tipos de dados**

Os tipos de dados especificam como números e caracteres serão armazenados e manipulados dentro de um programa. Os tipos de dados básicos do Python são:

1. **Números**

    1. **int** - inteiros
        
      **float** - Ponto flutuante

  **° Booleanos** - Assume os valores True ou False. Essencial quando comerçarmos a trabalhar com declarações condicionais


  **° Strings** - Sequencia de um ou mais caracteres que pode inclui letras, números e outros tipos de caracteres. Representa um texto.


  **° None** - Representa a ausencia de valor



**Números**

In [744]:
ano_atual = 2022

In [745]:
type(ano_atual) #função type, que passamos como argumento a variável e ele retorna que tipo de variável é

int

In [746]:
km_total = 44410.0

In [747]:
type(km_total)

float

**Booleanos**

In [748]:
zero_km = True

In [749]:
type(zero_km)

bool

In [750]:
zero_km = False

In [751]:
type(zero_km)

bool

**Strings**

In [752]:
nome = 'Jetta Variant'
nome

'Jetta Variant'

In [753]:
nome = "Jetta Variant"
nome

'Jetta Variant'

In [754]:
nome = 'Jetta "Variant"'
nome

'Jetta "Variant"'

In [755]:
nome = "Jetta 'Variant'"
nome

"Jetta 'Variant'"

In [756]:
carro = '''
Nome
Idade
Nota
'''

In [757]:
carro

'\nNome\nIdade\nNota\n'

In [758]:
type(carro)

str

**None**

In [759]:
quilometragem = None
quilometragem

In [760]:
type(quilometragem)

NoneType

**2.4 Conversão de tipos**

In [761]:
a = 10
b = 20
c = 'Python é '
d = 'legal'

In [762]:
type(a)

int

In [763]:
type(b)

int

In [764]:
type(c)

str

In [765]:
type(d)

str

In [766]:
c+d

'Python é legal'

In [767]:
#c + a 

**Conversões de tipo**

Funções int(), float(), str()

In [768]:
str(a) #Transforma uma variável que não é do tipo str em uma string

'10'

In [769]:
type(str(a))

str

In [770]:
c + str(a)

'Python é 10'

In [771]:
float(a)

10.0

In [772]:
type(float(a))

float

In [773]:
var = 3.141592

In [774]:
type(var)

float

In [775]:
int(var)

3

In [776]:
var = 3.99

In [777]:
int(var)

3

**Concatenando strings**

In [778]:
text = 'A quilometragem média do veículo é '
Km = 100000
Ano_atual = 2019
Ano_fabricacao = 1999

In [779]:
text + str( int( Km / (Ano_atual - Ano_fabricacao) ) ) + ' km' # foram realizadas duas conversões. Na primeira, o resultado da operação (float) é convertido em um inteiro (int) e depois, o dado do tipo inteiro é convertido em uma string (str).

'A quilometragem média do veículo é 5000 km'

**2.5 Indentação, comentários e formatação de strings**


**Indentação**

Na linguagem Python os programas são estrturados por meio de indentação. Em qualquer linguagem de programação a prática da indentação é bastante útil, facilitando a leitura e também a manutenção do código. Em Python a indentação não é somente uma questão de organização e estilo, mas sim um requisito da linguagem.

In [780]:
ano_atual = 2022
ano_fabricacao = 2022

if (ano_atual == ano_fabricacao):
  print('Verdadeiro')
else:
  print('Falso')

Verdadeiro


In [781]:
# Atalho para comentar Ctrl + ;

**Formatação de strings**

**str.format()**

In [782]:
print('Olá, {}!'.format('Gabriel')) #Na chave entra o parametro que está sendo passado para o .format, que no caso é 'Gabriel', substitui {} pelo o que tem no .format

Olá, Gabriel!


In [783]:
print('Olá, {}! Este é seu acesso de número {}'.format('Gabriel', 21))

Olá, Gabriel! Este é seu acesso de número 21


In [784]:
print('Olá, {nome}! Este é seu acesso de número {acessos}'.format(nome = 'Gabriel', acessos = 47))

Olá, Gabriel! Este é seu acesso de número 47


**f-Strings**

In [785]:
nome = 'Gabriel'
acessos = 47

In [786]:
print(f'Olá, {nome}! Este é seu acesso de número {acessos}') #Usando o f-Strings, basta só colocar o f na frente

Olá, Gabriel! Este é seu acesso de número 47


**3. Trabalhando com Listas**


**3.1 Criando listas**

Listas são sequencias mutáveis que são utilizadas para armazenar coleções de itens, geralmente homogeneos. Podem ser construídas de várias formas:

- Utilizando um par de colchetes: [], [1]
- Utilizando um par de colchetes com itens separados por vírgulas: [1, 2, 3]

In [787]:
Acessorios = ['Rodas de liga', 'Travas elétricas', 'Piloto automático', 'Bancos de couro', 'Ar condicionado', 'Sensor de estacionamento', 'Sensor crepuscular', 'Sensor de chuva']

In [788]:
Acessorios

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [789]:
type(Acessorios) #Tipo list

list

**Lista com tipos de dados variados**

In [790]:
Carro_1 = ['Jetta Variant', 'Motor 4.0 Turbo', 2003, 44410.0, False, ['Rodas de liga', 'Travas elétricas', 'Piloto automático'], 88078.64] # ['Rodas de liga', 'Travas elétricas', 'Piloto automático'] lista dentro de outra, quando digitarmos Carro_1[5], retornará ['Rodas de liga', 'Travas elétricas', 'Piloto automático'] na posição 5 que é uma outra lista
Carro_2 = ['Passat', 'Motor Diesel', 1991, 5712.0, False, ['Central multimídia', 'Teto panoramico', 'Freios ABS'], 106161.94]

In [791]:
Carro_1

['Jetta Variant',
 'Motor 4.0 Turbo',
 2003,
 44410.0,
 False,
 ['Rodas de liga', 'Travas elétricas', 'Piloto automático'],
 88078.64]

In [792]:
Carro_2

['Passat',
 'Motor Diesel',
 1991,
 5712.0,
 False,
 ['Central multimídia', 'Teto panoramico', 'Freios ABS'],
 106161.94]

In [793]:
Carros = [Carro_1, Carro_2] #Duas listas dentro de uma

In [794]:
Carros

[['Jetta Variant',
  'Motor 4.0 Turbo',
  2003,
  44410.0,
  False,
  ['Rodas de liga', 'Travas elétricas', 'Piloto automático'],
  88078.64],
 ['Passat',
  'Motor Diesel',
  1991,
  5712.0,
  False,
  ['Central multimídia', 'Teto panoramico', 'Freios ABS'],
  106161.94]]

In [795]:
Carro_1[5]

['Rodas de liga', 'Travas elétricas', 'Piloto automático']

In [796]:
Carro_2[5]

['Central multimídia', 'Teto panoramico', 'Freios ABS']

**3.2 Operações com listas**


**x in A** 

x é o elemento e A é a lista

Retorna **True** se um elemento da lista A for igual a x.

In [797]:
Acessorios

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [798]:
'Rodas de liga' in Acessorios #Rodas de liga está dentro de Acessorios? True

True

In [799]:
'4x4' in Acessorios #4x4 está dentro de Acessorios? False

False

In [800]:
'Rodas de liga' not in Acessorios #Rodas de liga não está em Acessorios? False

False

In [801]:
'4x4' not in Acessorios #4x4 não está dentro de Acessorios? True

True

**A + B**

Concatena as listas A e B

In [802]:
A = ['Rodas de liga', 'Travas elétricas', 'Piloto automático', 'Bancos de couro']
B = ['Ar condicionado', 'Sensor de estacionamento', 'Sensor crepuscular', 'Sensor de chuva']

In [803]:
A

['Rodas de liga', 'Travas elétricas', 'Piloto automático', 'Bancos de couro']

In [804]:
B

['Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [805]:
C = A + B

In [806]:
C

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

**len(A)**

Tamanho da lista A

In [807]:
len(Acessorios) #Tamanho da lista Acessorios

8

**3.3 Seleções em listas**

**A[i]**

Retorna o i-ésimo item da lista A.

Observação: Listas tem indexação com origem no zero

In [808]:
Acessorios

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [809]:
Acessorios[0]

'Rodas de liga'

In [810]:
Acessorios[1]

'Travas elétricas'

In [811]:
Acessorios[-1] # - 1, acesso o último item da lista, -2 o penultimo item da lista e assim sucesivamente...

'Sensor de chuva'

In [812]:
Carros

[['Jetta Variant',
  'Motor 4.0 Turbo',
  2003,
  44410.0,
  False,
  ['Rodas de liga', 'Travas elétricas', 'Piloto automático'],
  88078.64],
 ['Passat',
  'Motor Diesel',
  1991,
  5712.0,
  False,
  ['Central multimídia', 'Teto panoramico', 'Freios ABS'],
  106161.94]]

In [813]:
Carros[0]

['Jetta Variant',
 'Motor 4.0 Turbo',
 2003,
 44410.0,
 False,
 ['Rodas de liga', 'Travas elétricas', 'Piloto automático'],
 88078.64]

In [814]:
Carros[0][0] #Acessa a lista 1 e depois o primeiro item dessa lista

'Jetta Variant'

In [815]:
Carros[0][-2]

['Rodas de liga', 'Travas elétricas', 'Piloto automático']

In [816]:
Carros[0][-2][1]

'Travas elétricas'

**A[i:j**

Retorna a lista A do índice i até o j. Neste fatiamento o elemento com índice i é **incluído** e o elemento com índice j **não é incluído** no resultado

In [817]:
Carros

[['Jetta Variant',
  'Motor 4.0 Turbo',
  2003,
  44410.0,
  False,
  ['Rodas de liga', 'Travas elétricas', 'Piloto automático'],
  88078.64],
 ['Passat',
  'Motor Diesel',
  1991,
  5712.0,
  False,
  ['Central multimídia', 'Teto panoramico', 'Freios ABS'],
  106161.94]]

In [818]:
Carros[0][-2][0:2]

['Rodas de liga', 'Travas elétricas']

In [819]:
Acessorios

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [820]:
Acessorios[2:5]

['Piloto automático', 'Bancos de couro', 'Ar condicionado']

In [821]:
Acessorios[2:6]

['Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento']

In [822]:
Acessorios[2:]

['Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [823]:
Acessorios[:5]

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado']

Considere a seguinte lista:

In [824]:
letras = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H']

In [825]:
letras[0:2]

['A', 'B']

In [826]:
letras[:2]

['A', 'B']

In [827]:
letras[-3:]

['F', 'G', 'H']

In [828]:
letras[:-3]

['A', 'B', 'C', 'D', 'E']

In [829]:
carros = [
    [
        'Jetta Variant',
        'Motor 4.0 Turbo',
        2003,
        False,
        ['Rodas de liga', 'Travas elétricas', 'Piloto automático']
    ],
    [
        'Passat',
        'Motor Diesel',
        1991,
        True,
        ['Central multimídia', 'Teto panorâmico', 'Freios ABS']
    ]
]

In [830]:
carros

[['Jetta Variant',
  'Motor 4.0 Turbo',
  2003,
  False,
  ['Rodas de liga', 'Travas elétricas', 'Piloto automático']],
 ['Passat',
  'Motor Diesel',
  1991,
  True,
  ['Central multimídia', 'Teto panorâmico', 'Freios ABS']]]

In [831]:
carros[1][-2]

True

In [832]:
carros[1][3]

True

In [833]:
carros[1][-1][1]

'Teto panorâmico'

In [834]:
carros[1][-1][-2]

'Teto panorâmico'

In [835]:
carros[0][-1]

['Rodas de liga', 'Travas elétricas', 'Piloto automático']

**3.4 Métodos de listas**

**A.sort()**

Ordena a lista A

In [836]:
Acessorios = ['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [837]:
Acessorios

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Sensor de estacionamento',
 'Sensor crepuscular',
 'Sensor de chuva']

In [838]:
Acessorios.sort()
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

**A.append(x)**

Adiciona o elemento x no final da lista A

In [839]:
Acessorios.append('4 x 4')
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas',
 '4 x 4']

**A.pop(i)**

Remove e retorna o elemento de índice i da lista A.

Observação: Por default(padrão) o método pop() remove e retorna o último elemento de uma lista

In [840]:
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas',
 '4 x 4']

In [841]:
Acessorios.pop() #Remove o ultimo item da lista e retorna o que removeu

'4 x 4'

In [842]:
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

In [843]:
Acessorios.pop(3) #Remove o item na indexação 3 e retorna o que removeu

'Rodas de liga'

In [844]:
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

**A.copy()**

Cria uma cópia da lista A

Observação: O mesmo resultado pode ser obtido com o seguinte código:

A[:]

In [845]:
Acessorios_2 = Acessorios
Acessorios_2

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

In [846]:
Acessorios_2.append('4 x 4')
Acessorios_2

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas',
 '4 x 4']

In [847]:
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas',
 '4 x 4']

In [848]:
Acessorios.pop()

'4 x 4'

In [849]:
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

In [850]:
Acessorios_2

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

In [851]:
Acessorios_2 = Acessorios.copy() #Cria uma cópia do Acessorios e armazena em Acessorios_2. O parametro A[:] tb copia, ex: Acessorios_2 = Acessorios[:]
Acessorios_2

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

In [852]:
Acessorios_2.append('4 x 4')
Acessorios_2

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas',
 '4 x 4']

In [853]:
Acessorios

['Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Sensor crepuscular',
 'Sensor de chuva',
 'Sensor de estacionamento',
 'Travas elétricas']

In [854]:
Acessorios_atv = [
    'Rodas de liga', 
    'Travas elétricas', 
    'Piloto automático',
    'Bancos de couro',
    'Ar condicionado'
]

In [855]:
Acessorios_atv

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado']

In [856]:
Acessorios_atv.append('Airbag')

In [857]:
Acessorios_atv

['Rodas de liga',
 'Travas elétricas',
 'Piloto automático',
 'Bancos de couro',
 'Ar condicionado',
 'Airbag']

In [858]:
Acessorios_atv.sort()

In [859]:
Acessorios_atv

['Airbag',
 'Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga',
 'Travas elétricas']

In [860]:
Acessorios_atv.pop()

'Travas elétricas'

In [861]:
Acessorios_atv

['Airbag',
 'Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga']

In [862]:
Acessorios_atv.append('Vidros elétricos')

In [863]:
Acessorios_atv

['Airbag',
 'Ar condicionado',
 'Bancos de couro',
 'Piloto automático',
 'Rodas de liga',
 'Vidros elétricos']

**4. Estruturas de repetição e condicionais**

**4.1 Instrução for**
    
    Formatado padrão

        for <variavel> in <colecao>
            <instrucoes>